presto-native-execution/presto_cpp/main/Announcer.cpp

```diff
@@ -166,10 +174,24 @@ void Announcer::scheduleNext() {
   if (stopped_) {
     return;
   }
+
+  auto timeToSleepMs = frequencyMsMax_;
+  if (failedAttempts_ > 0) {
```

Comment:
Retrying infinitely doesn't look clean. We can cap number of failedAttempts - so there is only one codepath which does the inifinite retries.

We can also skip jitter here and use the manually added jitter in below comment. I'll leave this to your judgment